framework/helpers/BaseHtml.php

```diff
@@ -884,9 +884,12 @@ public static function checkboxList($name, $selection = null, $items = [], $opti
             $name .= '[]';
         }
 
-        $formatter = isset($options['item']) ? $options['item'] : null;
-        $itemOptions = isset($options['itemOptions']) ? $options['itemOptions'] : [];
-        $encode = !isset($options['encode']) || $options['encode'];
+        $formatter = ArrayHelper::remove($options, 'item');
+        $itemOptions = ArrayHelper::remove($options, 'itemOptions', []);
+        $encode = ArrayHelper::remove($options, 'encode', true);
+        $separator = ArrayHelper::remove($options, 'separator', PHP_EOL);
```

Comment:
PHP_EOL is specially useful when you want your code to be multiplataform and you will save the output to a file for example a cache file.

I can change it if you want http://stackoverflow.com/questions/4975411/when-do-i-use-php-eol-instead-of-n-and-vice-versa-ajax-jquery-client-problem